src/latch_cli/services/launch/launch_v2.py

```diff
@@ -0,0 +1,165 @@
+
+    python_interface_with_defaults: Union[dict[str, tuple[type, Any]], None] = None
+    for v in flyte_interface_types.values():
+        description: dict[str, Any] = json.loads(v.description)
+        if description.get("idx") != 0:
+            continue
+
+        raw_python_interface_with_defaults = description.get("__workflow_meta__", {}).get("meta", {}).get("python_interface")
+        if raw_python_interface_with_defaults is not None:
+            python_interface_with_defaults = dill.loads(base64.b64decode(raw_python_interface_with_defaults))  # noqa: S301
+            break
+
+    if python_interface_with_defaults is None:
+        raise ValueError("No python interface found -- re-register workflow with latch version >= 2.62.0 in workflow Dockerfile")
+
+    for k, v in python_interface_with_defaults.items():
+        if k in params:
+            continue
+
+        if v[1] is not None:
+            params[k] = v[1]
+            continue
+
+        t = v[0]
+        if get_origin(t) is Union and type(None) in get_args(t):
+            params[k] = None
+        else:
+            raise ValueError(f"Required parameter '{k}' not provided in params")
+
+    ctx = FlyteContextManager.current_context()
+    assert ctx is not None
+
+    fixed_literals = translate_inputs_to_literals(
+        ctx,
+        incoming_values=params,
+        flyte_interface_types=flyte_interface_types,
+        native_types={
+            k: v[0] for k, v in python_interface_with_defaults.items()
+        },
+    )
+
+    return launch_workflow(target_account_id, wf_id, {k: gpjson.MessageToDict(v.to_flyte_idl()) for k, v in fixed_literals.items()})
+
+
+def launch_workflow(target_account_id: str, wf_id: str, params: dict[str, Any]) -> int:
+    """Launch the workflow of given id with parameter map.
+
+    Return True if success, raises appropriate exceptions on failure.
+    """
+    response = tinyrequests.post(
+        config.api.execution.create,
+        headers={
+            "Authorization": get_auth_header(),
+            "User-Agent": (  # this is from a 3 yr old comment which says this was needed or the endpoint would hang sometimes?
+                "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like"
+                " Gecko) Chrome/72.0.3626.119 Safari/537.36"
+            ),
+        },
+        json={
+            "workflow_id": wf_id,
+            "params": params,
+            "ws_account_id": target_account_id,
+        },
+    )
+
+    try:
+        response_data = response.json()
+    except JSONDecodeError as e:
+        raise RuntimeError(f"Could not parse response as JSON: ({response.status_code}) {response}") from e
+
+    def extract_error_message(data: dict[str, Any]) -> str:
+        if "error" in data:
+            error = data["error"]
+            source = error.get("source", "unknown")
+
+            error_data = error.get("data", {})
+            message = (
+                error_data.get("stderr") or
```

Comment:
truthy comparisons here

id rather this be a sequence of `if message is None: message = ...`